src/server.js

```diff
@@ -47,16 +48,23 @@ app.use(
 
 app.use(setUser);
 
-// When route starts with "/api", use apiRoutes to handle
+// Public routes
 app.use("/", defaultRoute); // Lộc: Sửa route để khỏi trùng
 app.use("/article", articleRoutes); // Lộc: Thêm route còn thiếu
-app.use("/homepage", homepageRoute);
-app.use("/list", homepageRoute);
 
+// Global auth middleware
```

Comment:
If you use this mdw above /homepage and /list, guest users (who have no account) will have to login to read

Reply:
Protected routes should be `/writer`, `/editor`, `/profile` (writer and editor have another protection layer using role middleware)

Others can be accessed without user logged in

Reply:
I’m working on this one, please wait a bit.